### PR TITLE
python: Fix fslist1 FileNotFoundError crash

### PR DESCRIPTION
**Pull request by martinpitt**
Trying to list a nonexisting directory crashed the bridge. Report a    
proper channel problem instead.    
    
Add unit tests.    

----

- [x] Builds on top of PR #18285

This fixes the last [cockpit-machines failure](https://github.com/cockpit-project/cockpit-machines/pull/935). I ran `TestMachinesFilesystems.testBasicSystemConnection` with this fix and it passes now.

### CODE REVIEW

**Review by KKoukiou — APPROVED** [review]
Looks good to me and that's straightforward but since it's the first time I review pbridge code let's wait for one more review.

File: src/cockpit/channels/filesystem.py
Comment (allisonkarlitskaya) [inline src/cockpit/channels/filesystem.py:80]: I guess in these two cases, including the formatted strerror is somewhat redundant.

Also: in all cases, including the filename would probably be helpful.

**Review by allisonkarlitskaya — APPROVED** [review]
That said, this is an obvious improvement, so feel free to merge it.

File: src/cockpit/channels/filesystem.py
Comment (martinpitt) [reply src/cockpit/channels/filesystem.py:80]: The file name is part of the standard exception message actually:

```
❱❱❱ cpf open fslist1 path=dir1/ watch=False : sleep 1 | cpy |cat

{
  "channel": "ch1",
  "command": "close",
  "message": "[Errno 2] No such file or directory: 'dir1/'",
  "problem": "not-found"
}
```
I think that's good enough?

**Pull request merged**
